cmd/influxd/restore/restore.go

```diff
@@ -319,37 +630,68 @@ func (cmd *Command) unpackFile(tr *tar.Reader, fileName string) error {
 	if err != nil {
 		return err
 	}
-	defer ff.Close()
 
 	if _, err := io.Copy(ff, tr); err != nil {
+		if err := ff.Close(); err != nil {
+			return err
+		}
 		return err
 	}
 
+	if err := ff.Close(); err != nil {
+		return err
+	}
 	return nil
 }
 
 // printUsage prints the usage message to STDERR.
 func (cmd *Command) printUsage() {
 	fmt.Fprintf(cmd.Stdout, `Uses backups from the PATH to restore the metastore, databases,
-retention policies, or specific shards. The InfluxDB process must not be
-running during a restore.
+retention policies, or specific shards.  Legacy mode requires the instance to be stopped before running, and will wipe
+	all databases from the system (e.g., for disaster recovery).  The improved online mode requires the instance to be running,
+	and the database name used must not already exist.
+
+Usage: influxd restore [-enterprise] [flags] PATH
 
-Usage: influxd restore [flags] PATH
+The default mode consumes files in an OSS only file format. PATH is a directory containing the backup data
 
+Options:
     -metadir <path>
             Optional. If set the metastore will be recovered to the given path.
     -datadir <path>
             Optional. If set the restore process will recover the specified
-            database, retention policy or shard to the given directory.
+            destinationDatabase, retention policy or shard to the given directory.
     -database <name>
-            Optional. Required if no metadir given. Will restore the database
+            Optional. Required if no metadir given. Will restore the database's
             TSM files.
     -retention <name>
-            Optional. If given, database is required. Will restore the retention policy's
+            Optional. If given, -database is required. Will restore the retention policy's
             TSM files.
     -shard <id>
-            Optional. If given, database and retention are required. Will restore the shard's
+            Optional. If given, -database and -retention are required. Will restore the shard's
             TSM files.
+	-online
+	        Optional. If given, the restore will be done using the new process, detailed below.  All above legacy
+	        arguments are ignored.
+
+The -enterprise restore mode consumes files in an improved format that includes a file manifest.  PATH should indicate
```

Comment:
If we're going to call this `enterprise`, it should mimic the enterprise restore, accepting either a single manifest or a directory containing multiple manifests (as created by incremental backups)